service/collector/cluster_transition.go

```diff
@@ -0,0 +1,94 @@
+package collector
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/label"
+	aeconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ClusterTransitionTimeConfig struct {
+	CtrlClient ctrlclient.Client
+	Logger     micrologger.Logger
+}
+
+type ClusterTransitionTime struct {
+	ctrlClient ctrlclient.Client
+	logger     micrologger.Logger
+}
+
+var (
+	clusterTransitionCreateDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(MetricsNamespace, "cluster", "create_transition"),
+		"Latest cluster creation transition.",
+		[]string{
+			"cluster_id",
+			"release_version",
+		},
+		nil,
+	)
+)
+
+func NewClusterTransitionTime(config ClusterTransitionTimeConfig) (*ClusterTransitionTime, error) {
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CtrlClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	u := &ClusterTransitionTime{
+		ctrlClient: config.CtrlClient,
+		logger:     config.Logger,
+	}
+
+	return u, nil
+}
+
+func (u *ClusterTransitionTime) Collect(ch chan<- prometheus.Metric) error {
+	ctx := context.Background()
+	clusters := &capiv1alpha3.ClusterList{}
+	{
+		err := u.ctrlClient.List(ctx, clusters, ctrlclient.InNamespace(metav1.NamespaceAll))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	for _, cluster := range clusters.Items {
+		releaseVersion, ok := cluster.Labels[label.ReleaseVersion]
+		if !ok {
+			u.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("Cluster %#q has no %#q label. Skipping", cluster.Name, label.ReleaseVersion))
+			continue
+		}
+
+		if conditions.IsFalse(&cluster, aeconditions.CreatingCondition) {
+			creatingLastTransition := conditions.GetLastTransitionTime(&cluster, aeconditions.CreatingCondition)
+			ch <- prometheus.MustNewConstMetric(
+				clusterTransitionCreateDesc,
+				prometheus.GaugeValue,
+				cluster.CreationTimestamp.Sub(creatingLastTransition.Time).Seconds(),
+				cluster.Name,
+				releaseVersion,
+			)
+		}
```

Comment:
Does this emit the cluster creation time metric on every collection for the life of the cluster?

Reply:
Yes. AFAIK the `Creating` condition will be `false` once the cluster is created and it will stay that way for the life of the cluster.